instrumentation/jmx-metrics/javaagent/README.md

```diff
@@ -221,6 +221,54 @@ Thus, the above definitions will create several metrics, named `my.kafka.streams
 
 The metric descriptions will remain undefined, unless they are provided by the queried MBeans.
 
+### State Metrics
+
+Some JMX attributes expose current state as a non-numeric MBean attribute, in order to capture those as metrics it is recommended to use the special `state` metric type.
+For example, with Tomcat connector, the `Catalina:type=Connector,port=*` MBean has `stateName` (of type `String`), we can define the following rule:
+
+```yaml
+---
+rules:
+  - bean: Catalina:type=Connector,port=*
+    mapping:
+      stateName:
+        type: state
+        metric: tomcat.connector.state
+        metricAttribute:
+          port: param(port)
+          state:
+            ok: STARTED
+            failed: [STOPPED,FAILED]
+            degraded: '*'
+```
```

Comment:
While this example is nice, some readers might get confused by the word `state` appearing on 3 lines (235, 236, and 239). The meaning is different, but it might not be obvious to the readers.

Reply:
I agree let's remove any ambiguity when we can, I'll rename the metric `tomcat.connector` and the metric attribute to `connector_state`, do you think that will be enough here ?

Reply:
Yes, this will be great!